features/netsocket/SocketAddress.cpp

```diff
@@ -122,6 +122,7 @@ static void ipv6_from_address(uint8_t *bytes, const char *addr)
     // Move suffix to end
     memmove(&shorts[NSAPI_IPv6_BYTES/2-suffix], &shorts[0],
             suffix*sizeof(uint16_t));
+    memset(&shorts[0], 0, suffix*sizeof(uint16_t));
```

Comment:
@c1728p9, the value of `suffix` is bounded by this loop:
https://github.com/geky/mbed/blob/16c4da03e62fc2367b33be8bb9358c28804fffb3/features/netsocket/SocketAddress.cpp#L88

I don't think there is much value to adding an assert at this granularity, there are many other points of failure in this function.

@EduardPon, `memset` and friends can be optimized quite heavily by the compiler, usually emitting word-aligned assignments. The loop may be quicker, but without a test to check which is faster I just went with the higher-level operation.